core/validation/checks.py

```diff
@@ -11,121 +11,137 @@ class Check(ABC):
+    mapping: dict
+    mapped_column: int
+    mapped_row: int
+
+    def __init__(
+        self,
+        sheet: str,
+        column: int,
+        row: int,
+        expected_values: tuple,
+        error_message: str,
+        mapping: dict,
+        mapped_column: int,
+        mapped_row: int,
+    ):
+        super().__init__(sheet, column, row, expected_values, error_message)
+        self.mapping = mapping
+        self.mapped_column = mapped_column
+        self.mapped_row = mapped_row
+
+    def get_expected_values(self, workbook: dict[str, pd.DataFrame]) -> tuple:
+        value_to_map = str(workbook[self.sheet].iloc[self.mapped_column][self.mapped_row]).strip()
+        return self.mapping.get(value_to_map, [])
+
+    def run(self, workbook: dict[str, pd.DataFrame]) -> tuple[bool, str]:
+        result = self.get_actual_value(workbook) in self.get_expected_values(workbook)
+        return result, self.error_message
 
 
 class AuthorisationCheck(DynamicCheck):
     """
-    A dynamic check which uses authorization, verifying if the user is submitting data for the correct place names
-    and fund types associated with their account.
+    A dynamic check which uses authorisation, verifying if the user is submitting data for the correct place names or
+    fund types associated with their account.
 
     Though empty auth values are handled by the code here, submissions with auth values set to None are no longer
     permitted for round 4 data, and will raise a validation error in initial_validate.py whenever this is the case.
 
-    Parameters:
+    Attributes:
         Inherits attributes from the DynamicCheck class.
-        auth (dict or None): Authorization information to be set using the set_auth method.
+        auth_dict (dict): A dictionary of authorisation values for the user.
+        auth_type (str): The type of authorisation to check, e.g. "Place Names" or "Fund Types".
+
+    Methods:
+        set_auth_dict(auth_dict: dict | None):
+            Set the authorisation dictionary for the user.
+        substitute_error_message(actual_value: str, expected_values: tuple[str]) -> str:
+            Substitute the actual and expected values into the error message.
     """
 
-    auth: dict | None
+    auth_dict: dict | None
+    auth_type: str
 
-    def set_auth(self, auth: dict | None):
-        self.auth = auth
+    def __init__(self, sheet: str, column: int, row: int, expected_values: tuple, error_message: str, auth_type: str):
+        super().__init__(sheet, column, row, expected_values, error_message)
+        self.auth_type = auth_type
 
-    def calculate_expected_values(self, workbook: dict[str, pd.DataFrame]) -> tuple:
-        auth_type = self._dynamic_params["auth_type"]
-        expected_values = self.auth[auth_type] if self.auth else ()
-        self.expected_values = expected_values
+    def set_auth_dict(self, auth_dict: dict | None):
+        self.auth_dict = auth_dict
```

Comment:
This would also mutate the Check between ingests. We could pass `auth` to `AuthorisationCheck.run` I suppose but it's unfortunate that it would give it a different signature to the other classes..

Reply:
Will discuss this with Bobbi tomorrow morning and try to find an elegant solution. Will also discuss the comment relating to tests above